src/main/java/org/motechproject/prevac/service/impl/VisitRescheduleServiceImpl.java

```diff
@@ -255,22 +249,17 @@ private Time calculateEndTime(Time startTime) {
         return new Range<>(minDate, maxDate);
     }
 
-    private LocalDate getVaccinationDate(Visit visit, Boolean boosterRelated, Boolean thirdVaccinationRelated) {
+    private LocalDate getVaccinationDate(Visit visit, Boolean boosterRelated) {
         if (boosterRelated) {
             return visit.getSubject().getBoosterVaccinationDate();
         } else {
             return visit.getSubject().getPrimerVaccinationDate();
         }
     }
 
-    private Boolean isBoosterRelated(VisitType visitType, List<String> boosterRelatedMessages, Long stageId) {
+    private Boolean isBoosterRelated(VisitType visitType, List<String> boosterRelatedVisits, Long stageId) {
         String campaignName = getCampaignNameWithStage(visitType, stageId);
-        return boosterRelatedMessages.contains(campaignName);
-    }
-
-    private Boolean isThirdVaccinationRelated(VisitType visitType, List<String> thirdVaccinationRelatedMessages, Long stageId) {
-        String campaignName = getCampaignNameWithStage(visitType, stageId);
-        return thirdVaccinationRelatedMessages.contains(campaignName);
+        return boosterRelatedVisits.contains(campaignName);
```

Comment:
You're not adding stage id to the visit type in prevac config, so remove the campaign name and use just display name of the visit (you can also remove the getCampaignNameWithStage method then)